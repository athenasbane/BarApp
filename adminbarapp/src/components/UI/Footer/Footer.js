import React from 'react';
import { Grid, Button, makeStyles } from '@material-ui/core';

const useStyles = makeStyles({
  root: {
    width: '100%',
    position: 'fixed',
    left: 0,
    bottom: 0,
    backgroundColor: '#ccc',
    display: 'flex',
    justifyContent: 'center',
  },
  button: {
    width: '100%',
    textAlign: 'center',
  },
});

const Footer = () => {
  const classes = useStyles();
  return (
    <Grid className={classes.root} container item>
      <Button className={classes.button} variant="contained">
        Save
      </Button>
    </Grid>
  );
};

export default Footer;
