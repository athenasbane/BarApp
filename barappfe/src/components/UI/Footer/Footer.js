import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles, Button, Grid } from '@material-ui/core';
import { connect } from 'react-redux';
import Checkout from '../../Checkout/Checkout';

const useStyles = makeStyles({
  root: {
    width: '100%',
    position: 'fixed',
    left: 0,
    bottom: 0,
    backgroundColor: '#ccc',
    display: 'flex',
    justifyContent: 'center',
  },
  button: {
    width: '100%',
    textAlign: 'center',
  },
  border: {
    borderRight: '1px solid black',
  },
});

export const Footer = ({ order }) => {
  const classes = useStyles();
  const [open, setOpen] = React.useState(false);
  const checkoutClickHandler = () => {
    setOpen(!open);
  };

  return (
    <div className={classes.root}>
      <Checkout open={open} checkoutClickHandler={checkoutClickHandler} />
      <Grid container direction="row">
        <Grid className={classes.border} item xs={6}>
          <Button
            className={classes.button}
            disabled={order.length <= 0}
            color="secondary"
            onClick={checkoutClickHandler}
          >
            Basket&nbsp;({order.length})
          </Button>
        </Grid>
        <Grid item xs={6}>
          <Button
            className={classes.button}
            disabled={order.length <= 0}
            color="secondary"
            onClick={checkoutClickHandler}
          >
            CHECKOUT
          </Button>
        </Grid>
      </Grid>
    </div>
  );
};

Footer.propTypes = {
  order: PropTypes.arrayOf(PropTypes.object),
};

const mapStateToProps = (state) => ({
  order: state.order.orderData,
});

export default connect(mapStateToProps)(Footer);
