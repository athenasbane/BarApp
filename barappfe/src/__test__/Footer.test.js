import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { Footer } from '../components/UI/Footer/Footer';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore();
const store = mockStore();

describe('Footer Tests', () => {
    test('renders correctly', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <Footer />
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
}); 