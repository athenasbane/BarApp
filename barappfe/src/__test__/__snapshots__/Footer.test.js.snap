// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Tests renders correctly 1`] = `ShallowWrapper {}`;
